src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, (e) => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function resolves the promise only once per event. If you expect multiple events to be handled, consider using a different approach, such as attaching the event listener outside of the promise or using a loop to handle multiple events.